roofit/roofitcore/src/RooAbsCollection.cxx

```diff
@@ -710,30 +710,41 @@ RooAbsCollection* RooAbsCollection::selectByAttrib(const char* name, Bool_t valu
 }
 
 
-
-
 ////////////////////////////////////////////////////////////////////////////////
 /// Create a subset of the current collection, consisting only of those
 /// elements that are contained as well in the given reference collection.
-/// The caller is responsible for deleting the returned collection
+/// Returns `true` only if something went wrong.
+/// The complement of this function is getParameters().
+/// \param[in] refColl The collection to check for common elements.
+/// \param[out] outColl Output collection.
 
-RooAbsCollection* RooAbsCollection::selectCommon(const RooAbsCollection& refColl) const
+bool RooAbsCollection::selectCommon(const RooAbsCollection& refColl, RooAbsCollection& outColl) const
```

Comment:
Why not a collection as return value (copy elision)?

Reply:
We often have patterns like:
```C++
for(auto const& otherSet : otherSets) {
    RooArgSet * sel = ourSet.selectCommon(otherSet);
    delete sel;
}
```
while a function with returning a RooArgSet with copy-elision is convenient, it means the memory for the underlying std::vector has to be newly allocated in each iteration.

That's why we should definitely have functions with output parameters. Maybe we can also add a version with return parameter in the future when we hit a case where it can really save us many lines of code, but I don't think we need it for this PR.

Another problem is that we'd have to come up with a separate name, because the function signature would only differ by the return type from the original `selectCommon` that returns a pointer.

Reply:
I don't see the memory allocation issue, as it should also happen with sets that you put on the stack before calling the function.
I understand, however, that return-value overloading isn't possible in c++. 🙂

Reply:
I mean you can reduce the same set, for example:
```C++
RooArgSet common;
for(auto const& otherSet : otherSets) {
    ourSet.selectCommon(otherSet, comon);
}
```

It's not important for this PR, but for other places like RooProdPdf this will come in handy